protocol/response.go

```diff
@@ -109,3 +131,25 @@ func WriteResponse(w io.Writer, apiVersion int16, correlationID int32, msg Messa
 
 	return err
 }
+
+const (
+	tlsAlertByte byte = 0x15
+)
+
+// looksLikeUnexpectedTLS returns true if the size passed in resemble
+// the TLS alert message that is returned to a client which sends
+// an invalid ClientHello message.
+func looksLikeUnexpectedTLS(size int32) bool {
+	var sizeBytes [4]byte
+	binary.BigEndian.PutUint32(sizeBytes[:], uint32(size))
+
+	if sizeBytes[0] != tlsAlertByte {
+		return false
+	}
+	version := int(sizeBytes[1])<<8 | int(sizeBytes[2])
+
+	if version <= tls.VersionTLS13 && version >= tls.VersionTLS10 {
+		return true
+	}
+	return false
```

Comment:
Could this be simplified?
```suggestion
	return version <= tls.VersionTLS13 && version >= tls.VersionTLS10
```